### PR TITLE
Feature/7 adding rg template to folder.

### PR DESCRIPTION
**Pull request by curumoLuin**
Added LinkedTemplate. Seems to be working but not sure if correctly passing parameters. Review please.

### CODE REVIEW

File: arm-templates/subscription.json [hunk 1]
```diff
@@ -12,7 +12,9 @@
         }
     },
     "variables": {
-        "rgName": "[concat('rg-music-lake-', parameters('subPrefix')) ]"
+        "rgName": "[concat('rg-music-lake-', parameters('subPrefix')) ]",
+        "rgNameMusicLake": "[concat('rg-music-lake-', parameters('subPrefix')) ]",
+        "templateUrlRgMusicLake": "https://raw.githubusercontent.com/paweltajs/azure-data-lake-infra/feature/7-adding-rg-template-to-folder/arm-templates/resources/rg_music_lake.json"
```
Comment (paweltajs) [inline arm-templates/subscription.json:17]: We should pass path to root repository folder as parameter. Later on we should use this basePath to concat it with path to particular file.

File: arm-templates/subscription.json [hunk 2]
```diff
@@ -12,7 +12,9 @@
         }
     },
     "variables": {
-        "rgName": "[concat('rg-music-lake-', parameters('subPrefix')) ]"
+        "rgName": "[concat('rg-music-lake-', parameters('subPrefix')) ]",
+        "rgNameMusicLake": "[concat('rg-music-lake-', parameters('subPrefix')) ]",
```
Comment (paweltajs) [inline arm-templates/subscription.json:16]: Duplicate variable, they represent the same value. Lets make it **rg_music_lake** variable.

File: arm-templates/subscription.json [hunk 3]
```diff
@@ -21,6 +23,29 @@
...
+                "mode": "Incremental",
+                "expressionEvaluationOptions": {
+                    "scope": "outer"
+                },
+                "templateLink": {
+                    "uri": "[variables('templateUrlRgMusicLake')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "location": {
```
Comment (paweltajs) [inline arm-templates/subscription.json:41]: Make it oneliner if there is only one child.

File: arm-templates/subscription.json [hunk 4]
```diff
@@ -21,6 +23,29 @@
...
+                        "value": "[parameters('location')]"
+                    },
+                    "rgName": {
```
Comment (paweltajs) [inline arm-templates/subscription.json:44]: Make it oneliner if there is only one child. Use rg_music_lake as a name,

File: arm-templates/subscription.json [hunk 5]
```diff
@@ -21,6 +23,29 @@
...
             "location": "[parameters('location')]",
             "properties": {}
+        },
+        {
+            "type": "Microsoft.Resources/deployments",
+            "apiVersion": "2020-10-01",
+            "name": "linkedTemplate",
+            "properties": {
+                "mode": "Incremental",
+                "expressionEvaluationOptions": {
```
Comment (paweltajs) [inline arm-templates/subscription.json:33]: Make it oneliner if there is only one child.
Example:
"expressionEvaluationOptions": { scope": "outer" },

File: arm-templates/resources/rg_music_lake.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "westeurope"
```
Comment (paweltajs) [inline arm-templates/resources/rg_music_lake.json:7]: Use defaultValue only at subscription level. All nested templates should require to pass parameters.

File: arm-templates/resources/rg_music_lake.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        },
+        "rgName": {
+            "type": "string",
+            "defaultValue": "nestedParam"
```
Comment (paweltajs) [inline arm-templates/resources/rg_music_lake.json:11]: Use defaultValue only at subscription level. All nested templates should require to pass parameters.

**Review by paweltajs — CHANGES_REQUESTED** [review]

File: arm-templates/subscription.json (hunk 2 above)
Comment (curumoLuin) [reply arm-templates/subscription.json:16]: Well aware of that. Uploaded it just to check template linking. Most of errors and trash due to that. Appreciate all feedback non the less.

File: arm-templates/resources/rg_music_lake.json (hunk 7 above)
Comment (curumoLuin) [reply arm-templates/resources/rg_music_lake.json:11]: Also, just for testing actual outcome.

File: arm-templates/resources/rg_music_lake.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    "parameters": {
+        "location": {
+            "type": "string"
+        },
+        "rgName": {
+            "type": "string"
+        }
+    },
+    "variables": {    },
+    "resources": [
```
Comment (paweltajs) [inline arm-templates/resources/rg_music_lake.json:13]: Leave resources node empty.

File: arm-templates/subscription.json [hunk 9]
```diff
@@ -9,10 +9,16 @@
         "subPrefix": {
             "type": "string",
             "defaultValue": "cm002-dev"
+        },
+        "rootDir": {
+            "type": "string",
+            "defaultValue": "https://raw.githubusercontent.com/paweltajs/azure-data-lake-infra/"
```
Comment (paweltajs) [inline arm-templates/subscription.json:15]: Include branch name in rootDir parameter.

**Review by paweltajs — CHANGES_REQUESTED** [review]

**Review by paweltajs — APPROVED** [review]
We need to discuss linked templates on next meeting.

**Pull request merged**